lib/seahorn/BvOpSem2.cc

```diff
@@ -3182,6 +3186,45 @@ Optional<APInt> Bv2OpSem::agg(Type *aggTy,
   return res;
 }
 
+/// Crab stuff
+// Build Crab CFG, run crab analyses, etc.
+void Bv2OpSem::mkCrabAnalysis(const llvm::Module &M,
+                              seadsa::GlobalAnalysis &dsa,
+                              TargetLibraryInfoWrapperPass &tli) {
+
+  std::unique_ptr<clam::HeapAbstraction> heap_abs(
+      new clam::SeaDsaHeapAbstraction(M, dsa));
+
+  // -- Set parameters for CFG
+  clam::CrabBuilderParams cfg_builder_params;
+  LOG("bmc-crab", cfg_builder_params.print_cfg = true;);
+  // Note: default precision is NUM
+  cfg_builder_params.interprocedural = true;
+
+  m_cfg_builder_man.reset(new clam::CrabBuilderManager(cfg_builder_params, tli,
+                                                       std::move(heap_abs)));
+
+  // -- Initialize crab for solving ranges
+  m_crab_rng_solver.reset(new clam::InterGlobalClam(M, *m_cfg_builder_man));
+
+  /// Set Crab parameters
+  clam::AnalysisParams aparams;
+  aparams.dom = clam::CrabDomain::INTERVALS;
+  aparams.run_inter = true;
+  aparams.check = clam::CheckerKind::NOCHECKS;
+  /// Run the Crab analysis
+  clam::ClamGlobalAnalysis::abs_dom_map_t assumptions;
+  // aparams.print_invars = true;
+  m_crab_rng_solver->analyze(aparams, assumptions);
```

Comment:
I would run analysis on module enter event, not here

Reply:
Agreed. Will change my design.